pkg/connector/event_log.go

```diff
@@ -3,73 +3,91 @@ package connector
 	"github.com/conductorone/baton-sdk/pkg/types/resource"
 	oktaSDK "github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/okta/okta-sdk-golang/v2/okta/query"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func parseTarget(logEvent *oktaSDK.LogEvent) (*oktaSDK.LogTarget, bool) {
-	targetAppType := "AppInstance"
+const (
+	usageFilter           = `eventType eq "user.authentication.sso" and actor.type eq "User" and target.type eq "AppInstance"`
+	groupMembershipFilter = `eventType eq "group.user_membership.add" and target.type eq "UserGroup"`
+)
+
+func targetMap(logEvent *oktaSDK.LogEvent) (map[string][]*oktaSDK.LogTarget, error) {
+	rv := make(map[string][]*oktaSDK.LogTarget)
 	for _, target := range logEvent.Target {
-		if target.Type == targetAppType {
-			return target, true
-		}
+		rv[target.Type] = append(rv[target.Type], target)
 	}
-	return nil, false
+	return rv, nil
 }
 
-func (connector *Okta) listOktaSSOEvents(ctx context.Context, earliestEvent *timestamppb.Timestamp, pToken *pagination.StreamToken) ([]*oktaSDK.LogEvent, *oktaSDK.Response, error) {
+func (connector *Okta) createQueryParams(earliestEvent *timestamppb.Timestamp, pToken *pagination.StreamToken, filters ...string) *query.Params {
 	qp := queryParams(pToken.Size, pToken.Cursor)
 	if earliestEvent != nil {
 		qp.Since = earliestEvent.AsTime().Format(time.RFC3339)
 	}
-	qp.Filter = `eventType eq "user.authentication.sso" and actor.type eq "User" and target.type eq "AppInstance"`
 
-	logs, resp, err := connector.client.LogEvent.GetLogs(ctx, qp)
-	if err != nil {
-		return nil, nil, err
+	if len(filters) == 0 {
+		return qp
+	} else if len(filters) == 1 {
+		qp.Filter = filters[0]
+		return qp
 	}
 
-	return logs, resp, nil
+	filter := strings.Join(filters, ") or (")
+	qp.Filter = fmt.Sprintf(`(%s)`, filter)
+
+	return qp
 }
 
 func (connector *Okta) ListEvents(
 	ctx context.Context,
 	earliestEvent *timestamppb.Timestamp,
 	pToken *pagination.StreamToken,
 ) ([]*v2.Event, *pagination.StreamState, annotations.Annotations, error) {
-	logs, resp, err := connector.listOktaSSOEvents(ctx, earliestEvent, pToken)
+	qp := connector.createQueryParams(earliestEvent, pToken, groupMembershipFilter, usageFilter)
+
+	logs, resp, err := connector.client.LogEvent.GetLogs(ctx, qp)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
 	rv := make([]*v2.Event, 0, len(logs))
 	for _, log := range logs {
-		parsedTarget, isAppInstance := parseTarget(log)
-		if !isAppInstance {
-			continue
-		}
-		userTrait, err := resource.NewUserTrait(resource.WithEmail(log.Actor.AlternateId, true))
+		targetMap, err := targetMap(log)
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
-		rv = append(rv, &v2.Event{
+		event := &v2.Event{
 			Id:         log.Uuid,
 			OccurredAt: timestamppb.New(*log.Published),
-			Event: &v2.Event_UsageEvent{
+		}
+		switch log.EventType {
+		case "user.authentication.sso":
+			if len(targetMap["AppInstance"]) != 1 {
```

Comment:
maybe leave a comment explaining this check here. Why is it not > 0 as opposed to != 1